tesseract_process_managers/src/taskflow_generators/graph_taskflow.cpp

```diff
@@ -47,159 +52,68 @@ TaskflowContainer GraphTaskflow::generateTaskflow(TaskInput input, TaskflowVoidF
-      }
-      else if (node.edges[0].dest_channel == DestinationChannel::DONE_CALLBACK &&
-               node.edges[1].dest_channel == DestinationChannel::PROCESS_NODE)
-      {
-        if (node.edges[0].src_channel == SourceChannel::ON_SUCCESS &&
-            node.edges[1].src_channel == SourceChannel::ON_FAILURE)
-          tasks[src].precede(tasks[static_cast<std::size_t>(node.edges[1].dest)], container.outputs[1]);
-        else if (node.edges[1].src_channel == SourceChannel::ON_SUCCESS &&
-                 node.edges[0].src_channel == SourceChannel::ON_FAILURE)
-          tasks[src].precede(container.outputs[1], tasks[static_cast<std::size_t>(node.edges[1].dest)]);
-      }
-      else if (node.edges[0].dest_channel == DestinationChannel::ERROR_CALLBACK &&
-               node.edges[1].dest_channel == DestinationChannel::PROCESS_NODE)
-      {
-        if (node.edges[0].src_channel == SourceChannel::ON_SUCCESS &&
-            node.edges[1].src_channel == SourceChannel::ON_FAILURE)
-          tasks[src].precede(tasks[static_cast<std::size_t>(node.edges[1].dest)], container.outputs[0]);
-        else if (node.edges[1].src_channel == SourceChannel::ON_SUCCESS &&
-                 node.edges[0].src_channel == SourceChannel::ON_FAILURE)
-          tasks[src].precede(container.outputs[0], tasks[static_cast<std::size_t>(node.edges[1].dest)]);
-      }
-      else if (node.edges[0].dest_channel == DestinationChannel::DONE_CALLBACK &&
-               node.edges[1].dest_channel == DestinationChannel::ERROR_CALLBACK)
-      {
-        if (node.edges[0].src_channel == SourceChannel::ON_SUCCESS &&
-            node.edges[1].src_channel == SourceChannel::ON_FAILURE)
-          tasks[src].precede(container.outputs[0], container.outputs[1]);
-        else if (node.edges[1].src_channel == SourceChannel::ON_SUCCESS &&
-                 node.edges[0].src_channel == SourceChannel::ON_FAILURE)
-          tasks[src].precede(container.outputs[1], container.outputs[0]);
-      }
-      else if (node.edges[0].dest_channel == DestinationChannel::ERROR_CALLBACK &&
-               node.edges[1].dest_channel == DestinationChannel::DONE_CALLBACK)
-      {
-        if (node.edges[0].src_channel == SourceChannel::ON_SUCCESS &&
-            node.edges[1].src_channel == SourceChannel::ON_FAILURE)
-          tasks[src].precede(container.outputs[1], container.outputs[0]);
-        else if (node.edges[1].src_channel == SourceChannel::ON_SUCCESS &&
-                 node.edges[0].src_channel == SourceChannel::ON_FAILURE)
-          tasks[src].precede(container.outputs[0], container.outputs[1]);
-      }
-      else
-      {
-        throw std::runtime_error("Invalid Edges for process index: " + std::to_string(src_idx));
-      }
+      tasks.at(i).precede(tasks.at(static_cast<std::size_t>(idx)));
+    }
+
+    // If no edges exist for the current node, then make sure this node precedes the done task
+    if (node.edges.empty())
```

Comment:
Should we throw and exception here and make the developer fix?

Reply:
I don't think so. Users don't have access to the done and error tasks, so they can't explicitly add them to the graph. The implicit assumption right now is that connections are automatically made to the done and error tasks for leaf nodes

Reply:
OK, so the developer would need to visualize the graph to insure they correctly built the graph.